### ISSUE TITLE
JPG not resizing, just black

### ISSUE THREAD

**Issue by tillkruss** [kean/Nuke#782]
I'm noticing some `.jpg` result in a black image when resized. Is that an issue with the file, Nuke, or my code?

```swift
import Nuke
import NukeUI
import SwiftUI

struct Test: View {
    let url = "https://image.tmdb.org/t/p/original/f0JpsMQ9oEjKBD66Ky3qK3z7LGT.jpg"

    var body: some View {
        LazyImage(request: imageRequest(url)) { state in
            if let image = state.image {
                let _ = print("success")
                image.resizable()
            } else if state.error != nil {
                let _ = print(state.error)
            } else {
                Text("loading")
            }
        }
    }

    func imageRequest(_ urlString: String?) -> ImageRequest {
        let url = URL(string: urlString!)
        let request = URLRequest(url: url!, timeoutInterval: 5)

        return ImageRequest(
            urlRequest: request,
            processors: [
                .resize(size: CGSize(width: 325, height: 488))
            ]
        )
    }
}

#Preview {
    Test()
}
```

**Comment by kean** [comment]
Hey, thanks for the report, @tillkruss. Yes, it is a defect in the framework.

**RCA**
(for future reference)

- The input image is grayscale (8 bpp pixel format) and with no alpha channel
- The framework tried tries to create `CGContext` with the grayscale CS, 8 bpp and `CGImageGetAlphaInfo.nonSkipLast`, an [unsupported combination](https://developer.apple.com/library/archive/documentation/GraphicsImaging/Conceptual/drawingwithquartz2d/dq_context/dq_context.html#//apple_ref/doc/uid/TP30001066-CH203-BCIBHHBB)
- The framework expects the `CGContext` init to fail and fall back to a safe combination of parameters supported on all Apple devices. But `CGContext` does get created anyway and fails to render images correctly.

Fixed in [v12.7](https://github.com/kean/Nuke/releases/tag/12.7.0).

**Comment by tillkruss** [comment]
Thanks a lot! ❤️
